tests/test_information_from_events.py

```diff
@@ -0,0 +1,464 @@
+    for (A, B, clAB, clBA) in trustlines:
+        contract.functions.setAccountDefaultInterests(
+            accounts[A], accounts[B], clAB, clBA, False, 0, 0, 0, 0
+        ).transact()
+    return contract
+
+
+@pytest.fixture()
+def currency_network_with_pending_interests(
+    web3, currency_network_contract_with_trustlines, accounts, chain
+):
+    """
+    returns a currency network with updated balance accruing interests over one year
+    the interests are not collected yet.
+    """
+    for (A, B, clAB, clBA) in trustlines:
+        currency_network_contract_with_trustlines.functions.transfer(
+            accounts[B], 10, 0, [accounts[B]], b""
+        ).transact({"from": accounts[A]})
+
+    timestamp = web3.eth.getBlock("latest").timestamp
+    timestamp += 3600 * 24 * 365
+
+    chain.time_travel(timestamp)
+    chain.mine_block()
+    return currency_network_contract_with_trustlines
+
+
+@pytest.fixture()
+def currency_network_with_different_interests(
+    web3, currency_network_contract_with_trustlines, accounts, chain
+):
+    """
+    returns a currency network with updated balance accruing interests over one year
+    the interests are not collected yet
+    """
+    for (A, B, clAB, clBA) in trustlines:
+        currency_network_contract_with_trustlines.functions.transfer(
+            accounts[B], clAB // 10, 0, [accounts[B]], b""
+        ).transact({"from": accounts[A]})
+
+    timestamp = web3.eth.getBlock("latest").timestamp
+    timestamp += (3600 * 24 * 365) * 3
+
+    chain.time_travel(timestamp)
+    chain.mine_block()
+    return currency_network_contract_with_trustlines
+
+
+def calculate_interests(
+    balance: int, delta_time_in_seconds: int, highest_order: int = 15
+) -> int:
+    """
+    This function was taken from the relay server, for a proof of the correctness of this calculation see
+    `relay/tests/unit/network_graph/test_interests.py`: test_interests_calculation_gives_same_result_as_smart_contracts
+    """
+    SECONDS_PER_YEAR = 60 * 60 * 24 * 365
+    INTERESTS_DECIMALS = 2
+    internal_interest_rate = 1000
+
+    intermediate_order = balance
+    interests = 0
+    # Calculate compound interests using taylor approximation
+    for order in range(1, highest_order + 1):
+        intermediate_order = int(
+            intermediate_order
+            * internal_interest_rate
+            * delta_time_in_seconds
+            / (SECONDS_PER_YEAR * 100 * 10 ** INTERESTS_DECIMALS * order)
+        )
+
+        if intermediate_order == 0:
+            break
+        interests += intermediate_order
+
+    return interests
+
+
+def get_interests_for_trustline(currency_network_contract, a, b):
+    """Get all balance changes of a trustline because of interests"""
+    balance_updates = get_balance_updates_for_trustline(currency_network_contract, a, b)
+
+    timestamps = [
+        currency_network_contract.web3.eth.getBlock(balance_update[0][0])["timestamp"]
```

Comment:
Am I supposed to know that `balance_update[0][0]` is the block_number corresponding to the balance_update?

Reply:
Solved by using `balance_update_event["blockNumber"]`